server/liveblog/blogs/tasks.py

```diff
@@ -36,37 +37,92 @@ def publish_embed(blog_id, api_host=None, theme=None):
     return superdesk.upload.url_for_media(file_id)
 
 
-def delete_embed(blog_id):
+def delete_embed(blog_id, theme=None, output=None):
     check_media_storage()
-    file_path = get_blog_path(blog_id)
+    blog = get_resource_service('client_blogs').find_one(req=None, _id=blog_id)
+    public_urls = blog.get('public_urls', {'output': {}, 'theme': {}})
+    if output:
+        output_id = str(output.get('_id'))
+        public_url = public_urls['output'][output_id]
+        public_urls['output'].pop(output_id)
+    elif theme:
+        public_url = public_urls['theme'][theme]
+        public_urls['theme'].pop(theme)
+    else:
+        for output_url in public_urls['outputs']:
+            app.media.delete(output_url)
+        for theme_url in public_urls['theme']:
+            app.media.delete(theme_url)
+        public_url = blog.get('public_url')
+        public_urls = {}
+
     # Remove existing file.
-    app.media.delete(file_path)
+    app.media.delete(public_url)
+    get_resource_service('blogs').system_update(blog_id, {'public_urls': public_urls}, blog)
 
 
-def _publish_blog_embed_on_s3(blog_id, safe=True):
+def _publish_blog_embed_on_s3(blog_id, theme=None, output=None, safe=True):
```

Comment:
Should 'safe' not be false by default? I think you should have to explicitly suppress or handle the exception. Otherwise you risk silent failure.